api/envoy/type/metadata/v2/metadata.proto

```diff
@@ -2,11 +2,13 @@ syntax = "proto3";
 
 package envoy.type.metadata.v2;
 
+import "udpa/annotations/migrate.proto";
 import "validate/validate.proto";
 
 option java_package = "io.envoyproxy.envoy.type.metadata.v2";
 option java_outer_classname = "MetadataProto";
 option java_multiple_files = true;
+option (udpa.annotations.file_migrate).move_to_package = "envoy.type.metadata.v3";
```

Comment:
Why are we force moving this and other similar things if we don't need to?

Reply:
The metadata matcher docs were referring to v2 types that got moved in v3. So, this was the easiest way to solve it. I think in these specific cases, it was good to move for two reasons:
1. These are foundational types, so it's a bit more consistent.
2. This didn't cause any transitive moves, so it had low impact on folks doing migration.

If you think these aren't compelling, I can look into some other solution, like allowing dual references in `:ref` and then preprocessing them, or even regenerating just the docs for shared protos, but keeping the protos the same. The latter isn't a ton of work, but up to you.

Reply:
I see, OK. That's fine.